Add valgrind ctest memcheck, add doc, and fix a ut

jw-msft opened the pull request:
- Add valgrind integrated testing via ctest -T memcheck
- Add valgrind memcheck to quick start and howto build docs
- Fix memleaks in file_info_utils_ut.cpp

JeffMill commented on src/diagnostics_component/utils/file_info_utils/tests/file_info_utils_ut.cpp:
```diff
@@ -38,7 +38,10 @@ TEST_CASE("FileInfoUtils_FillFileInfoWithNewestFilesInDir")
 
         CHECK(strcmp(sortedLogFileArray[0].fileName, testCandidateFileName) == 0);
+
+        free(sortedLogFileArray[0].fileName);
```
this test is CPP.  use unique_ptr

JeffMill replied:
i see we have :

calloc_wrapper
FreeDeleter (unused?)
std::unique_ptr<char, decltype(&free)> managed(ptr, free);
and i think other wrappers

we should converge on one.  and use it.

jw-msft replied:
I moved to using calloc_wrapper / cstr_wrapper and got rid of all the reset from ut .cpp files.

jw-msft commented:
```sh
~/src/iot-hub-device-update/out$ ctest -T memcheck
   Site: jwz4
   Build name: Linux-c++
Create new tag: 20230303-0016 - Experimental
Memory check project /home/jw/src/fork/iot-hub-device-update/out
        Start   1: DiagnosticsConfigUtils_Init
  1/122 MemCheck   #1: DiagnosticsConfigUtils_Init ..................................................   Passed    1.36 sec
        Start   2: FileInfoUtils_FillFileInfoWithNewestFilesInDir
  2/122 MemCheck   #2: FileInfoUtils_FillFileInfoWithNewestFilesInDir ...............................   Passed    1.34 sec
        Start   3: Download content
  3/122 MemCheck   #3: Download content .............................................................   Passed    1.36 sec
        Start   4: ReadDelimitedValueFromFile
  4/122 MemCheck   #4: ReadDelimitedValueFromFile ...................................................   Passed    1.35 sec
        Start   5: ARRAY_SIZE
  5/122 MemCheck   #5: ARRAY_SIZE ...................................................................   Passed    1.35 sec
        Start   6: ADUC_StringUtils_Trim
  6/122 MemCheck   #6: ADUC_StringUtils_Trim ........................................................   Passed    1.36 sec
        Start   7: ADUC_ParseUpdateType and atoui
  7/122 MemCheck   #7: ADUC_ParseUpdateType and atoui ...............................................   Passed    1.38 sec
        Start   8: atoul
  8/122 MemCheck   #8: atoul ........................................................................   Passed    1.33 sec
        Start   9: ADUC_StrNLen
  9/122 MemCheck   #9: ADUC_StrNLen .................................................................   Passed    1.35 sec
        Start  10: ADUC_StringFormat
 10/122 MemCheck  #10: ADUC_StringFormat ............................................................   Passed    1.35 sec
        Start  11: ConnectionStringUtils_DoesKeyExist
 11/122 MemCheck  #11: ConnectionStringUtils_DoesKeyExist ...........................................   Passed    1.32 sec
        Start  12: ConnectionStringUtils_GetValue_badoutarg
 12/122 MemCheck  #12: ConnectionStringUtils_GetValue_badoutarg .....................................   Passed    1.33 sec
        Start  13: ConnectionStringUtils_GetValue_key_missing
 13/122 MemCheck  #13: ConnectionStringUtils_GetValue_key_missing ...................................   Passed    1.30 sec
        Start  14: ConnectionStringUtils_GetValue_key_exists
 14/122 MemCheck  #14: ConnectionStringUtils_GetValue_key_exists ....................................   Passed    1.31 sec
        Start  15: ConnectionStringUtils_IsNestedEdge nullptr connection string returns false
 15/122 MemCheck  #15: ConnectionStringUtils_IsNestedEdge nullptr connection string returns false ...   Passed    1.31 sec
        Start  16: ConnectionStringUtils_IsNestedEdge returns false
 16/122 MemCheck  #16: ConnectionStringUtils_IsNestedEdge returns false .............................   Passed    1.38 sec
        Start  17: ConnectionStringUtils_IsNestedEdge returns true
 17/122 MemCheck  #17: ConnectionStringUtils_IsNestedEdge returns true ..............................   Passed    1.36 sec
        Start  18: ConnectionStringUtils_GetModuleIdFromConnectionString Success Case
 18/122 MemCheck  #18: ConnectionStringUtils_GetModuleIdFromConnectionString Success Case ...........   Passed    1.33 sec
        Start  19: ConnectionStringUtils_GetModuleIdFromConnectionString Failure Case
 19/122 MemCheck  #19: ConnectionStringUtils_GetModuleIdFromConnectionString Failure Case ...........   Passed    1.33 sec
        Start  20: ConnectionStringUtils_GetDeviceIdFromConnectionString Success Case
 20/122 MemCheck  #20: ConnectionStringUtils_GetDeviceIdFromConnectionString Success Case ...........   Passed    1.32 sec
        Start  21: ConnectionStringUtils_GetDeviceIdFromConnectionString Failure Case
 21/122 MemCheck  #21: ConnectionStringUtils_GetDeviceIdFromConnectionString Failure Case ...........   Passed    1.32 sec
        Start  22: ADUC_ConfigInfo_Init Functional Tests
 22/122 MemCheck  #22: ADUC_ConfigInfo_Init Functional Tests ........................................***Failed    1.43 sec
        Start  23: ADUC_ContractUtils_IsV1Contract
 23/122 MemCheck  #23: ADUC_ContractUtils_IsV1Contract ..............................................   Passed    1.31 sec
        Start  24: Base64 Encoding
 24/122 MemCheck  #24: Base64 Encoding ..............................................................   Passed    1.33 sec
        Start  25: Base64 Decoding
 25/122 MemCheck  #25: Base64 Decoding ..............................................................   Passed    1.34 sec
        Start  26: RSA Keys
 26/122 MemCheck  #26: RSA Keys .....................................................................   Passed    1.34 sec
        Start  27: Signature Verification
 27/122 MemCheck  #27: Signature Verification .......................................................***Failed    1.48 sec
        Start  28: RequestConnectionStringFromEISWithExpiry Functional Tests
 28/122 MemCheck  #28: RequestConnectionStringFromEISWithExpiry Functional Tests ....................***Failed    1.43 sec
        Start  29: ADUC_HashUtils_GetFileHash - SmallFile
 29/122 MemCheck  #29: ADUC_HashUtils_GetFileHash - SmallFile .......................................   Passed    1.44 sec
        Start  30: ADUC_HashUtils_IsValidFileHash - SmallFile
 30/122 MemCheck  #30: ADUC_HashUtils_IsValidFileHash - SmallFile ...................................   Passed    1.44 sec
        Start  31: ADUC_HashUtils_IsValidBufferHash
 31/122 MemCheck  #31: ADUC_HashUtils_IsValidBufferHash .............................................   Passed    1.38 sec
        Start  32: ADUC_HashUtils_GetShaVersionForTypeString
 32/122 MemCheck  #32: ADUC_HashUtils_GetShaVersionForTypeString ....................................   Passed    1.34 sec
        Start  33: ADUC_HashUtils_IsValidFileHash - LargeFile
 33/122 MemCheck  #33: ADUC_HashUtils_IsValidFileHash - LargeFile ...................................   Passed    3.88 sec
        Start  34: ADUC_HashUtils_GetFileHash - LargeFile
 34/122 MemCheck  #34: ADUC_HashUtils_GetFileHash - LargeFile .......................................   Passed    1.42 sec
        Start  35: Parse https_proxy (escaped)
 35/122 MemCheck  #35: Parse https_proxy (escaped) ..................................................   Passed    1.30 sec
        Start  36: Parse https_proxy
 36/122 MemCheck  #36: Parse https_proxy ............................................................   Passed    1.33 sec
        Start  37: Parse HTTPS_PROXY (uppper case)
 37/122 MemCheck  #37: Parse HTTPS_PROXY (uppper case) ..............................................   Passed    1.32 sec
        Start  38: Use https_proxy (lower case)
 38/122 MemCheck  #38: Use https_proxy (lower case) .................................................   Passed    1.32 sec
        Start  39: Parse username and password
 39/122 MemCheck  #39: Parse username and password ..................................................   Passed    1.33 sec
        Start  40: No port number
 40/122 MemCheck  #40: No port number ...............................................................   Passed    1.32 sec
        Start  41: Empty username
 41/122 MemCheck  #41: Empty username ...............................................................   Passed    1.32 sec
        Start  42: Empty password (supported)
 42/122 MemCheck  #42: Empty password (supported) ...................................................   Passed    1.32 sec
        Start  43: IsInstalled_test
 43/122 MemCheck  #43: IsInstalled_test .............................................................   Passed    1.39 sec
        Start  44: RemoveIsInstalledWhenEmpty
 44/122 MemCheck  #44: RemoveIsInstalledWhenEmpty ...................................................   Passed    1.28 sec
        Start  45: RemoveIsInstalledWhenNoMatch
 45/122 MemCheck  #45: RemoveIsInstalledWhenNoMatch .................................................   Passed    1.36 sec
        Start  46: RemoveIsInstalledTwice_test
 46/122 MemCheck  #46: RemoveIsInstalledTwice_test ..................................................   Passed    1.37 sec
        Start  47: RemoveInstalledCriteriaShouldRemoveDuplicates
 47/122 MemCheck  #47: RemoveInstalledCriteriaShouldRemoveDuplicates ................................   Passed    1.37 sec
        Start  48: RemoveInstalledCriteriaShouldRemoveDuplicatesAndSkipNonMatching
 48/122 MemCheck  #48: RemoveInstalledCriteriaShouldRemoveDuplicatesAndSkipNonMatching ..............   Passed    1.38 sec
        Start  49: PermissionUtils_VerifyFilemodeBit*
 49/122 MemCheck  #49: PermissionUtils_VerifyFilemodeBit* ...........................................   Passed    1.29 sec
        Start  50: VerifySJWK
 50/122 MemCheck  #50: VerifySJWK ...................................................................***Failed    1.53 sec
        Start  51: VerifyJWSWithKey
 51/122 MemCheck  #51: VerifyJWSWithKey .............................................................***Failed    1.50 sec
        Start  52: ADUC_FileEntity_Init
 52/122 MemCheck  #52: ADUC_FileEntity_Init .........................................................   Passed    1.34 sec
        Start  53: SanitizePathSegment
 53/122 MemCheck  #53: SanitizePathSegment ..........................................................   Passed    1.33 sec
        Start  54: apt-get fail
 54/122 MemCheck  #54: apt-get fail .................................................................   Passed    1.40 sec
        Start  55: hostname error
 55/122 MemCheck  #55: hostname error ...............................................................   Passed    1.36 sec
        Start  56: Invalid option - cp -1
 56/122 MemCheck  #56: Invalid option - cp -1 .......................................................   Passed    1.35 sec
        Start  57: VerifyProcessEffectiveGroup
 57/122 MemCheck  #57: VerifyProcessEffectiveGroup ..................................................   Passed    1.36 sec
        Start  58: VerifyProcessEffectiveUser
 58/122 MemCheck  #58: VerifyProcessEffectiveUser ...................................................***Failed    1.34 sec
        Start  59: TrimLeading
 59/122 MemCheck  #59: TrimLeading ..................................................................   Passed    1.33 sec
        Start  60: TrimTrailing
 60/122 MemCheck  #60: TrimTrailing .................................................................   Passed    1.33 sec
        Start  61: Trim
 61/122 MemCheck  #61: Trim .........................................................................   Passed    1.34 sec
        Start  62: Split
 62/122 MemCheck  #62: Split ........................................................................   Passed    1.36 sec
        Start  63: RemoveSurrounding
 63/122 MemCheck  #63: RemoveSurrounding ............................................................   Passed    1.33 sec
        Start  64: ADUC_SystemUtils_GetTemporaryPathName
 64/122 MemCheck  #64: ADUC_SystemUtils_GetTemporaryPathName ........................................   Passed    1.31 sec
        Start  65: ADUC_SystemUtils_ExecuteShellCommand
 65/122 MemCheck  #65: ADUC_SystemUtils_ExecuteShellCommand .........................................   Passed    1.37 sec
        Start  66: ADUC_SystemUtils_MkDirDefault
 66/122 MemCheck  #66: ADUC_SystemUtils_MkDirDefault ................................................   Passed    1.32 sec
        Start  67: ADUC_SystemUtils_MkDirRecursiveDefault
 67/122 MemCheck  #67: ADUC_SystemUtils_MkDirRecursiveDefault .......................................   Passed    1.34 sec
        Start  68: ADUC_SystemUtils_RmDirRecursive
 68/122 MemCheck  #68: ADUC_SystemUtils_RmDirRecursive ..............................................   Passed    1.33 sec
        Start  69: SystemUtils_ForEachDir
 69/122 MemCheck  #69: SystemUtils_ForEachDir .......................................................   Passed    1.39 sec
        Start  70: Initialization test
 70/122 MemCheck  #70: Initialization test ..........................................................   Passed    1.42 sec
        Start  71: Undefined update action
 71/122 MemCheck  #71: Undefined update action ......................................................   Passed    1.36 sec
        Start  72: Get Compatibility
 72/122 MemCheck  #72: Get Compatibility ............................................................***Failed    1.40 sec
        Start  73: Update Id
 73/122 MemCheck  #73: Update Id ....................................................................   Passed    1.39 sec
        Start  74: Child workflow uses fileUrls from parent
 74/122 MemCheck  #74: Child workflow uses fileUrls from parent .....................................***Failed    1.40 sec
        Start  75: Get update file by name
 75/122 MemCheck  #75: Get update file by name ......................................................***Failed    1.40 sec
        Start  76: Get update file by name - mixed case
 76/122 MemCheck  #76: Get update file by name - mixed case .........................................***Failed    1.39 sec
        Start  77: Add and remove children
 77/122 MemCheck  #77: Add and remove children ......................................................***Failed    1.40 sec
        Start  78: Set workflow result
 78/122 MemCheck  #78: Set workflow result ..........................................................***Failed    1.40 sec
        Start  79: Get update manifest version - 1.0
 79/122 MemCheck  #79: Get update manifest version - 1.0 ............................................   Passed    1.37 sec
        Start  80: Get update manifest version - 2.0
 80/122 MemCheck  #80: Get update manifest version - 2.0 ............................................   Passed    1.38 sec
        Start  81: Get update manifest version - 2
 81/122 MemCheck  #81: Get update manifest version - 2 ..............................................   Passed    1.38 sec
        Start  82: Get update manifest version - 3
 82/122 MemCheck  #82: Get update manifest version - 3 ..............................................   Passed    1.38 sec
        Start  83: Get update manifest version - x
 83/122 MemCheck  #83: Get update manifest version - x ..............................................   Passed    1.35 sec
        Start  84: Get update manifest version - empty
 84/122 MemCheck  #84: Get update manifest version - empty ..........................................   Passed    1.35 sec
        Start  85: Get update manifest version - missing
 85/122 MemCheck  #85: Get update manifest version - missing ........................................   Passed    1.37 sec
        Start  86: Mininum update manifest version check - 5
 86/122 MemCheck  #86: Mininum update manifest version check - 5 ....................................   Passed    1.38 sec
        Start  87: Mininum update manifest version check - 1
 87/122 MemCheck  #87: Mininum update manifest version check - 1 ....................................***Failed    1.52 sec
        Start  88: workflow_id_compare
 88/122 MemCheck  #88: workflow_id_compare ..........................................................   Passed    1.39 sec
        Start  89: workflow_isequal_id
 89/122 MemCheck  #89: workflow_isequal_id ..........................................................   Passed    1.38 sec
        Start  90: result success erc
 90/122 MemCheck  #90: result success erc ...........................................................***Failed    1.40 sec
        Start  91: Request workflow cancellation
 91/122 MemCheck  #91: Request workflow cancellation ................................................***Failed    1.41 sec
        Start  92: workflow_get_update_file with download handler
 92/122 MemCheck  #92: workflow_get_update_file with download handler ...............................***Failed    1.61 sec
        Start  93: workflow_get_update_file - upd metadata missing relatedFile URL
 93/122 MemCheck  #93: workflow_get_update_file - upd metadata missing relatedFile URL ..............***Failed    1.54 sec
        Start  94: MicrosoftDeltaDownloadHandlerUtils_ProcessRelatedFile Cache Miss
 94/122 MemCheck  #94: MicrosoftDeltaDownloadHandlerUtils_ProcessRelatedFile Cache Miss .............***Failed    1.58 sec
        Start  95: ADUC_SourceUpdateCacheUtils_CreateSourceUpdateCachePath
 95/122 MemCheck  #95: ADUC_SourceUpdateCacheUtils_CreateSourceUpdateCachePath ......................   Passed    1.34 sec
        Start  96: ADUC_SourceUpdateCacheUtils_MoveToUpdateCache
 96/122 MemCheck  #96: ADUC_SourceUpdateCacheUtils_MoveToUpdateCache ................................   Passed    1.53 sec
        Start  97: ADUC_SourceUpdateCacheUtils_PurgeOldestFromUpdateCache
 97/122 MemCheck  #97: ADUC_SourceUpdateCacheUtils_PurgeOldestFromUpdateCache .......................   Passed    1.67 sec
        Start  98: APT Parser Tests
 98/122 MemCheck  #98: APT Parser Tests .............................................................   Passed    1.38 sec
        Start  99: APT Parser Tests 2
 99/122 MemCheck  #99: APT Parser Tests 2 ...........................................................   Passed    1.38 sec
        Start 100: APT Parser AgentRestartRequired Test
100/122 MemCheck #100: APT Parser AgentRestartRequired Test .........................................   Passed    1.35 sec
        Start 101: APT Parser AgentRestartRequired(false) Test
101/122 MemCheck #101: APT Parser AgentRestartRequired(false) Test ..................................   Passed    1.36 sec
        Start 102: APT Parser AgentRestartRequired(du-agent package name) Test
102/122 MemCheck #102: APT Parser AgentRestartRequired(du-agent package name) Test ..................   Passed    1.36 sec
        Start 103: Download Succeeded 500
103/122 MemCheck #103: Download Succeeded 500 .......................................................***Failed    1.43 sec
        Start 104: Download Failed 12345
104/122 MemCheck #104: Download Failed 12345 ........................................................***Failed    1.44 sec
        Start 105: Download Catch-All 22222
105/122 MemCheck #105: Download Catch-All 22222 .....................................................   Passed    1.42 sec
        Start 106: Download - No Results - Succeeded 500
106/122 MemCheck #106: Download - No Results - Succeeded 500 ........................................***Failed    1.43 sec
        Start 107: Install Skipped 603
107/122 MemCheck #107: Install Skipped 603 ..........................................................   Passed    1.43 sec
        Start 108: Install Failed 333333
108/122 MemCheck #108: Install Failed 333333 ........................................................   Passed    1.41 sec
        Start 109: Install - No Results - Succeeded 600
109/122 MemCheck #109: Install - No Results - Succeeded 600 .........................................   Passed    1.40 sec
        Start 110: Apply failed 44444
110/122 MemCheck #110: Apply failed 44444 ...........................................................   Passed    1.44 sec
        Start 111: Apply - No Results - Succeeded 700
111/122 MemCheck #111: Apply - No Results - Succeeded 700 ...........................................   Passed    1.42 sec
        Start 112: Cancel failed 55555
112/122 MemCheck #112: Cancel failed 55555 ..........................................................   Passed    1.43 sec
        Start 113: Cancel - No Results - Succeeded 800
113/122 MemCheck #113: Cancel - No Results - Succeeded 800 ..........................................   Passed    1.43 sec
        Start 114: IsInstalled - installed
114/122 MemCheck #114: IsInstalled - installed ......................................................   Passed    1.43 sec
        Start 115: IsInstalled - notInstalled
115/122 MemCheck #115: IsInstalled - notInstalled ...................................................   Passed    1.43 sec
        Start 116: Backup Succeeded 1000
116/122 MemCheck #116: Backup Succeeded 1000 ........................................................   Passed    1.43 sec
        Start 117: Restore Succeeded 1100
117/122 MemCheck #117: Restore Succeeded 1100 .......................................................   Passed    1.45 sec
        Start 118: SWUpdate Prepare Arguments Test
118/122 MemCheck #118: SWUpdate Prepare Arguments Test ..............................................***Failed    1.53 sec
        Start 119: SWUpdate sample script --action-is-installed
119/122 MemCheck #119: SWUpdate sample script --action-is-installed .................................***Failed    1.57 sec
        Start 120: SWUpdate sample script --action-download
120/122 MemCheck #120: SWUpdate sample script --action-download .....................................***Failed    1.58 sec
        Start 121: SWUpdate sample script --action-apply
121/122 MemCheck #121: SWUpdate sample script --action-apply ........................................***Failed    1.57 sec
        Start 122: SWUpdate sample script --action-cancel
122/122 MemCheck #122: SWUpdate sample script --action-cancel .......................................***Failed    1.57 sec

79% tests passed, 26 tests failed out of 122

Total Test time (real) = 172.02 sec

The following tests FAILED:
	 22 - ADUC_ConfigInfo_Init Functional Tests (Failed)
	 27 - Signature Verification (Failed)
	 28 - RequestConnectionStringFromEISWithExpiry Functional Tests (Failed)
	 50 - VerifySJWK (Failed)
	 51 - VerifyJWSWithKey (Failed)
	 58 - VerifyProcessEffectiveUser (Failed)
	 72 - Get Compatibility (Failed)
	 74 - Child workflow uses fileUrls from parent (Failed)
	 75 - Get update file by name (Failed)
	 76 - Get update file by name - mixed case (Failed)
	 77 - Add and remove children (Failed)
	 78 - Set workflow result (Failed)
	 87 - Mininum update manifest version check - 1 (Failed)
	 90 - result success erc (Failed)
	 91 - Request workflow cancellation (Failed)
	 92 - workflow_get_update_file with download handler (Failed)
	 93 - workflow_get_update_file - upd metadata missing relatedFile URL (Failed)
	 94 - MicrosoftDeltaDownloadHandlerUtils_ProcessRelatedFile Cache Miss (Failed)
	103 - Download Succeeded 500 (Failed)
	104 - Download Failed 12345 (Failed)
	106 - Download - No Results - Succeeded 500 (Failed)
	118 - SWUpdate Prepare Arguments Test (Failed)
	119 - SWUpdate sample script --action-is-installed (Failed)
	120 - SWUpdate sample script --action-download (Failed)
	121 - SWUpdate sample script --action-apply (Failed)
	122 - SWUpdate sample script --action-cancel (Failed)
-- Processing memory checking output:
22/122 MemCheck: #22: ADUC_ConfigInfo_Init Functional Tests ........................................   Defects: 18
27/122 MemCheck: #27: Signature Verification .......................................................   Defects: 4
28/122 MemCheck: #28: RequestConnectionStringFromEISWithExpiry Functional Tests ....................   Defects: 6
50/122 MemCheck: #50: VerifySJWK ...................................................................   Defects: 3
51/122 MemCheck: #51: VerifyJWSWithKey .............................................................   Defects: 1
58/122 MemCheck: #58: VerifyProcessEffectiveUser ...................................................   Defects: 2
72/122 MemCheck: #72: Get Compatibility ............................................................   Defects: 1
74/122 MemCheck: #74: Child workflow uses fileUrls from parent .....................................   Defects: 1
75/122 MemCheck: #75: Get update file by name ......................................................   Defects: 1
76/122 MemCheck: #76: Get update file by name - mixed case .........................................   Defects: 1
77/122 MemCheck: #77: Add and remove children ......................................................   Defects: 1
78/122 MemCheck: #78: Set workflow result ..........................................................   Defects: 1
87/122 MemCheck: #87: Mininum update manifest version check - 1 ....................................   Defects: 3
90/122 MemCheck: #90: result success erc ...........................................................   Defects: 2
91/122 MemCheck: #91: Request workflow cancellation ................................................   Defects: 1
92/122 MemCheck: #92: workflow_get_update_file with download handler ...............................   Defects: 6
93/122 MemCheck: #93: workflow_get_update_file - upd metadata missing relatedFile URL ..............   Defects: 5
94/122 MemCheck: #94: MicrosoftDeltaDownloadHandlerUtils_ProcessRelatedFile Cache Miss .............   Defects: 6
103/122 MemCheck: #103: Download Succeeded 500 .......................................................   Defects: 4
104/122 MemCheck: #104: Download Failed 12345 ........................................................   Defects: 4
106/122 MemCheck: #106: Download - No Results - Succeeded 500 ........................................   Defects: 4
118/122 MemCheck: #118: SWUpdate Prepare Arguments Test ..............................................   Defects: 1
119/122 MemCheck: #119: SWUpdate sample script --action-is-installed .................................   Defects: 1
120/122 MemCheck: #120: SWUpdate sample script --action-download .....................................   Defects: 1
121/122 MemCheck: #121: SWUpdate sample script --action-apply ........................................   Defects: 1
122/122 MemCheck: #122: SWUpdate sample script --action-cancel .......................................   Defects: 1
MemCheck log files can be found here: ( * corresponds to test number)
/home/jw/src/fork/iot-hub-device-update/out/Testing/Temporary/MemoryChecker.*.log
Memory checking results:
Mismatched deallocation - 2
Memory Leak - 73
Uninitialized Memory Conditional - 5
        Errors while running CTest
```

JeffMill commented on README.md:
````diff
@@ -27,3 +27,46 @@ Device Update for IoT Hub features provide a powerful and flexible experience, i
+ctest
+```
+
+or, alternatively:
````
nit: i'd avoid mentioning ninja specifically here, and stick to cmake/ctest instructions.

JeffMill reviewed (approved):
added one suggestion
